giskard-server/src/main/java/ai/giskard/web/rest/controllers/ProjectController.java

```diff
@@ -86,6 +87,10 @@ public ProjectDTO getProject(@RequestParam(value = "id", required = false) Long
         } else {
             throw new IllegalArgumentException("Either project id or project key should be specified");
         }
+        if (project.getInspectionSettings() == null){
```

Comment:
FYI, "get" method shouldn't have side effects like saving something in the database.
Also I changed a bit the way how JSON objects get initialized while reading null from the database, here's a fix:
https://github.com/Giskard-AI/giskard/compare/1adce11d451a...9043c55d6d60

Reply:
ok that's really smart !